File: frontend/src/types/PartnerAssignment.ts
export interface PartnerAssignment {
  assignmentId: number,
  partnerId: number,
  safehouseId?: number | null,
  programArea: string,
  assignmentStart: string,
  assignmentEnd?: string | null,
  responsibilityNotes?: string | null,
  isPrimary: boolean,
  status: string,
}

File: backend/Havyn.API/Data/PartnerAssignment.cs
using System.ComponentModel.DataAnnotations;

namespace Havyn.Data
{
    public class PartnerAssignment
    {
        [Key]
        public int AssignmentId { get; set; }
        [Required]
        public int PartnerId { get; set; }
        public float? SafehouseId { get; set; }
        [Required]
        public string ProgramArea { get; set; }
        [Required]
        public DateOnly AssignmentStart { get; set; }
        public DateOnly? AssignmentEnd { get; set; }
        public string? ResponsibilityNotes { get; set; }
        [Required]
        public bool IsPrimary { get; set; }
        [Required]
        public string Status { get; set; }
    }
}
